game/src/ecs/components/xp/XPComponent.java

```diff
@@ -0,0 +1,118 @@
+package ecs.components.xp;
+
+import ecs.components.Component;
+import ecs.entities.Entity;
+
+public class XPComponent extends Component {
+
+    private static final long FORMULA_A = 100;
+    private static final long FORMULA_B = 10;
+    private long currentLevel;
+    private long currentXP;
+    private long lootXP = -1;
+    private ILevelUp callbackLevelUp;
+
+    /**
+     * Create a new XP-Component and add it to the associated entity
+     *
+     * @param entity associated entity
+     */
+    public XPComponent(Entity entity) {
+        super(entity);
+    }
+
+    /**
+     * Create a new XP-Component and add it to the associated entity
+     *
+     * @param entity associated entity
+     * @param levelUp callback for when the entity levels up
+     */
+    public XPComponent(Entity entity, ILevelUp levelUp) {
+        super(entity);
+        this.callbackLevelUp = levelUp;
+    }
+
+    /**
+     * Get the current level of the entity
+     *
+     * @return current level
+     */
+    public long getCurrentLevel() {
+        return currentLevel;
+    }
+
+    /**
+     * Set the current level of the entity
+     *
+     * @param currentLevel current level
+     */
+    public void setCurrentLevel(long currentLevel) {
+        this.currentLevel = currentLevel;
+    }
+
+    /**
+     * Add xp to the entity
+     *
+     * @param xp xp to add
+     */
+    public void addXP(long xp) {
+        this.currentXP += xp;
```

Comment:
wäre das nicht vereinfacht
`currentXP = Math.max(0,currentXP+xp);`

Außerdem: `this` bitte nur da wo notwendig

Und bitte in die javadoc schreiben, dass bei XP-Verlust maximal auf 0-XP verloren werden kann.

Reply:
Der Konsistent halber lass ich es aber weg.